### PR TITLE
feat: JsonSchema generated forms

### PR DESCRIPTION
**Pull request by ShubhranshuSanjeev**
## Problem
Describe the problem you are trying to solve here

## Solution
Provide a brief summary of your solution so that reviewers can understand your code

### CODE REVIEW

File: crates/frontend/Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,28 @@ name = "frontend"
...
+# # Should `wasm-opt` be used to further optimize the wasm binary generated after
+# # the Rust compiler has finished? Using `wasm-opt` can often further decrease
+# # binary size or do clever tricks that haven't made their way into LLVM yet.
+# #
+# # Configuration is set to `false` by default for the dev profile, but it can
+# # be set to an array of strings which are explicit arguments to pass to
+# # `wasm-opt`. For example `['-Os']` would optimize for size while `['-O4']`
+# # would execute very expensive optimizations passes
+# wasm-opt = ['-Os']
+
```
Comment (Datron) [inline crates/frontend/Cargo.toml:27]: Why add this?

File: crates/frontend/src/app.rs [hunk 2]
```diff
@@ -83,6 +83,7 @@ pub fn app(app_envs: Envs) -> impl IntoView {
             // sets the document title
             <Title text="Welcome to Superposition"/>
             <script type_="text/javascript">"__APP_ENVS=" {json!(app_envs).to_string()}</script>
+            <script src="https://pfau-software.de/json-viewer/dist/iife/index.js"></script>
```
Comment (Datron) [inline crates/frontend/src/app.rs:86]: Can we download this and host it locally? Rather risky URL if pfau software goes down or the company closes

File: crates/frontend/src/components/monaco_editor.rs [hunk 3]
```diff
@@ -37,22 +41,35 @@ pub fn monaco_editor(
             minimap_settings.set_enabled(Some(false));
             editor_settings.set_language(Some(language.to_string().as_str()));
             editor_settings.set_automatic_layout(Some(true));
-            editor_settings.set_value(Some(data_rs.get().as_str()));
+            editor_settings.set_font_family(Some("ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, \"Liberation Mono\", \"Courier New\", monospace"));
+            logging::log!("Monaco Init value {}", data);
+            editor_settings.set_value(Some(data.as_str()));
+            // editor_settings.set_value(Some(data_rs.get().as_str()));
```
Comment (Datron) [inline crates/frontend/src/components/monaco_editor.rs:47]: This might break functions UI

File: crates/frontend/src/components/monaco_editor.rs [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
 }
 
-pub type EditorModelCell = Rc<RefCell<Option<CodeEditor>>>;
+pub type EditorModelCell = Rc<Option<CodeEditor>>;
 
 #[component]
 pub fn monaco_editor(
-    node_id: &'static str,
+    #[prop(into)] node_id: String,
+    #[prop(into, default = String::new())] data: String,
```
Comment (Datron) [inline crates/frontend/src/components/monaco_editor.rs:20]: Why take data separately?

**Review by Datron — COMMENTED** [review]
Partial walk through, didn't review the whole thing

File: crates/frontend/src/app.rs (hunk 2 above)
Comment (ShubhranshuSanjeev) [reply crates/frontend/src/app.rs:86]: I think that would be better. Tbh, a big chunk of the js is just the themes, we can remove that too from the local version.

File: crates/frontend/src/components/monaco_editor.rs (hunk 4 above)
Comment (ShubhranshuSanjeev) [reply crates/frontend/src/components/monaco_editor.rs:20]: I want to remove the signals and use this `data` parameter, and handle any changes with `on_change` callback

File: crates/frontend/src/components/monaco_editor.rs (hunk 3 above)
Comment (ShubhranshuSanjeev) [reply crates/frontend/src/components/monaco_editor.rs:47]: Oh, i will make changes wherever this component is being used to utilize this implementation

File: crates/frontend/Cargo.toml (hunk 1 above)
Comment (ShubhranshuSanjeev) [reply crates/frontend/Cargo.toml:27]: Was trying different optimization strategies for wasm-pack, will remove this, or maybe we can keep it if we want to pin on some optimization profile

File: crates/frontend/src/app.rs (hunk 2 above)
Comment (ShubhranshuSanjeev) [reply crates/frontend/src/app.rs]: Moved to jsdeliver CDN url

File: crates/frontend/src/components/side_nav.rs [hunk 5]
```diff
@@ -64,8 +64,9 @@ pub fn side_nav(
     let location = use_location();
     let tenant_rs = use_context::<ReadSignal<String>>().unwrap();
     let tenant_ws = use_context::<WriteSignal<String>>().unwrap();
-    let (app_routes, set_app_routes) =
-        create_signal(create_routes(tenant_rs.get().as_str()));
+    let (app_routes, set_app_routes) = create_signal::<Vec<AppRoute>>(vec![]);
+
+    logging::log!("{:?}", app_routes.get());
```
Comment (sauraww) [inline crates/frontend/src/components/side_nav.rs:69]: Can we add some beautiful description here ?

**Review by sauraww — CHANGES_REQUESTED** [review]

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/app.rs [hunk 6]
```diff
@@ -38,9 +38,13 @@ pub fn app(app_envs: Envs) -> impl IntoView {
...
                     <Link rel="shortcut icon" type_="image/ico" href=favicon_href/>
                     <Link
-                        href="https://cdn.jsdelivr.net/npm/remixicon@3.5.0/fonts/remixicon.css"
+                        href="https://cdn.jsdelivr.net/npm/remixicon/fonts/remixicon.css"
                         rel="stylesheet"
                     />
+                    <script
+                        type_="text/javascript"
+                        src="https://cdn.jsdelivr.net/npm/@andypf/json-viewer@2.1.5/dist/iife/index.min.js"
+                    ></script>
```
Comment (Datron) [inline crates/frontend/src/app.rs:47]: In a later PR, lets download and self host these dependencies

**Review by Datron — APPROVED** [review]

File: crates/frontend/src/components/dropdown.rs [hunk 7]
```diff
@@ -74,30 +77,28 @@ where
...
+                    view! {
+                        <div class="mb-3">
+                            <label class="input input-bordered flex items-center gap-2 h-10">
+                                <i class="ri-search-line"></i>
+                                <input
+                                    type="text"
+                                    class="grow"
+                                    placeholder="Search"
+                                    name=name
+                                    value=String::new()
```
Comment (sauraww) [inline crates/frontend/src/components/dropdown.rs:90]:
```suggestion
                                    value={search_term.get_untracked()}
```

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/components/input.rs [hunk 8]
```diff
@@ -0,0 +1,462 @@
...
+pub fn monaco_input(
+    id: String,
+    class: String,
+    value: Value,
+    on_change: Callback<Value, ()>,
+    schema_type: SchemaType,
+) -> impl IntoView {
+    let id = store_value(id);
+    let schema_type = store_value(schema_type);
+    let (value_rs, value_ws) = create_signal(value);
```
Comment (sauraww) [inline crates/frontend/src/components/input.rs:253]:
```suggestion
       // Can we default to empty Json if no value is provided ?
    let default_value = Value::Object(serde_json::Map::new());
    let (value_rs, value_ws) = create_signal(value.unwrap_or(default_value.clone()));
```

File: crates/frontend/src/components/monaco_editor.rs [hunk 9]
```diff
@@ -37,22 +38,32 @@ pub fn monaco_editor(
...
-        }>
+        <div
+            id=node_id
+            class=styling
+            node_ref=editor_ref
+            on:keyup=move |_| {
+                match editor_rs.get().borrow() {
+                    Some(editor) => {
+                        logging::log!("here editor {:?}", editor.get_model().unwrap().get_value());
+                        on_change.call(editor.get_model().unwrap().get_value());
```
Comment (sauraww) [inline crates/frontend/src/components/monaco_editor.rs:61]: Can we handle it gracefully? If we don't get the value , we should not call on_change . 
Also , we should avoid using plain unwrap

File: crates/frontend/src/components/override_form.rs [hunk 10]
```diff
@@ -3,20 +3,116 @@ use std::collections::{HashMap, HashSet};
...
+                view! {
+                    <div class="badge badge-outline text-gray-400 font-medium text-xs">
+                        {r#type.to_string()}
+                    </div>
+                }
+            })
+            .collect_view(),
+        None => view! {}.into_view(),
+    }
+}
```
Comment (sauraww) [inline crates/frontend/src/components/override_form.rs:36]:
```suggestion
#[component]
fn type_badge(r#type: Option<SchemaType>) -> impl IntoView {
    r#type.map(|schema_type| match schema_type {
        SchemaType::Single(r#type) => view! {
            <div class="badge badge-outline text-gray-400 font-medium text-xs">
                {r#type.to_string()}
            </div>
        },
        SchemaType::Multiple(types) => types
            .iter()
            .map(|r#type| {
                view! {
                    <div class="badge badge-outline text-gray-400 font-medium text-xs">
                        {r#type.to_string()}
                    </div>
                }
            })
            .collect_view(),
    })
    .unwrap_or_default() 
}
```

**Review by sauraww — CHANGES_REQUESTED** [review]

**Review by sauraww — APPROVED** [review]

**Pull request merged**